src/System.CommandLine.Subsystems/ErrorReportingSubsystem.cs

```diff
@@ -1,24 +1,49 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.CommandLine.Parsing;
 using System.CommandLine.Subsystems;
 using System.CommandLine.Subsystems.Annotations;
 
 namespace System.CommandLine;
 
+/// <summary>
+/// Subsystem for reporting errors
+/// </summary>
+/// <remarks>
+/// This class, including interface, is likey to change as powderhouse continues
+/// </remarks>
 public class ErrorReportingSubsystem : CliSubsystem
 {
     public ErrorReportingSubsystem(IAnnotationProvider? annotationProvider = null)
         : base(ErrorReportingAnnotations.Prefix, SubsystemKind.ErrorReporting, annotationProvider)
     { }
 
-    // TODO: Stash option rather than using string
     protected internal override bool GetIsActivated(ParseResult? parseResult)
         => parseResult is not null && parseResult.Errors.Any();
 
+    // TODO: properly test execute directly when parse result is usable in tests
     protected internal override CliExit Execute(PipelineContext pipelineContext)
     {
-        pipelineContext.ConsoleHack.WriteLine("You have errors!");
+        var _ = pipelineContext.ParseResult
+            ?? throw new ArgumentNullException($"{nameof(pipelineContext)}.ParseResult");
```

Comment:
It is un-usual to have the name of an argument exception not directly map to the argument name of the method. Typically when it is a nested property we just use an ArgumentException with a message (NB: ArgumentException, and ArgumentNullException do not do their constructor parameters in the same order).

Also we likely need to localize the exception message.

```suggestion
            ?? throw new ArgumentException("The parse result has not been set", nameof(pipelineContext));
```